File: src/main/java/dev/lbuddyboy/flash/user/comparator/PunishmentDateComparator.java
package dev.lbuddyboy.flash.user.comparator;

import dev.lbuddyboy.flash.user.model.Punishment;
import dev.lbuddyboy.flash.user.model.UserPermission;

import java.util.Comparator;

public class PunishmentDateComparator implements Comparator<Punishment> {

    @Override
    public int compare(Punishment punishment, Punishment otherPunishment) {
        return Long.compare(punishment.getSentAt(), otherPunishment.getSentAt());
    }
}

File: src/main/java/dev/lbuddyboy/flash/user/comparator/PunishmentDateComparator.kt
package dev.lbuddyboy.flash.user.comparator

import dev.lbuddyboy.flash.user.model.Punishment

class PunishmentDateComparator : Comparator<Punishment> {
    override fun compare(punishment: Punishment, otherPunishment: Punishment): Int {
        return java.lang.Long.compare(punishment.sentAt, otherPunishment.sentAt)
    }
}